### PR TITLE
Add list of hooks and tools when fetching plugins

### PR DESCRIPTION
**Pull request by zAlweNy26**
# Description

In this way, you can visualize in the frontend or from a generic client, the used hooks by each plugin and their priority, for a better understanding of the flow of the plugins you installed. You can also see the tools created by the plugins.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by pieroit** [comment]
This is a good idea thanks
What about adding the tools?

**Comment by zAlweNy26** [comment]
I thought about that too, but... who cares about the tools names? The description of the plugin should already answer a hypothetical question like "how do I trigger this plugin?", or am I wrong?

File: core/cat/routes/plugins.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ async def get_available_plugins(
         # get manifest
         manifest = deepcopy(p.manifest) # we make a copy to avoid modifying the plugin obj
         manifest["active"] = p.id in active_plugins # pass along if plugin is active or not
+        manifest["used_hooks"] = [{ "name": hook.name, "priority": hook.priority } for hook in p.hooks]
+        manifest["used_tools"] = [tool.func.__name__ for tool in p.tools]
```
Comment (zAlweNy26) [inline core/cat/routes/plugins.py:44]: Also added tools, should the entry be called `used_tools`, `tools` or `created_tools`?

**Comment by zAlweNy26** [comment]
I was thinking about changing the data structure in this way, where the key is the priority:

```json
{
  "0": ["hook name 1", "hook name 2"],
  "1": ["hook name 3", "hook name 4"]
}
```

(Suggested by the Cat 😉)

**Comment by pieroit** [comment]
> I was thinking about changing the data structure in this way, where the key is the priority:
> 
> ```json
> {
>   "0": ["hook name 1", "hook name 2"],
>   "1": ["hook name 3", "hook name 4"]
> }
> ```
> 
> (Suggested by the Cat 😉)

The previous was better, because it makes no assumptions on ordering or indexing.
Two keys `tools` and `hooks` with a list of dicta inside

**Pull request merged**